e2e/actions/oci.go

```diff
@@ -1544,10 +1546,271 @@ func (c actionTests) actionOciOverlayExtfsPerms(t *testing.T) {
 	}
 }
 
+//nolint:maintidx
+func (c actionTests) actionOciBindImage(t *testing.T) {
+	e2e.EnsureOCISIF(t, c.env)
+	imageRef := "oci-sif:" + c.env.OCISIFPath
+
+	require.Command(t, "mkfs.ext3")
+	require.Command(t, "mksquashfs")
+	require.Command(t, "dd")
+
+	testdir, err := os.MkdirTemp(c.env.TestDir, "bind-image-")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	scratchDir := filepath.Join(testdir, "scratch")
+	if err := os.MkdirAll(filepath.Join(scratchDir, "bin"), 0o700); err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup := func(t *testing.T) {
+		if t.Failed() {
+			t.Logf("Not removing directory %s for test %s", testdir, t.Name())
+			return
+		}
+		err := os.RemoveAll(testdir)
+		if err != nil {
+			t.Logf("Error while removing directory %s for test %s: %#v", testdir, t.Name(), err)
+		}
+	}
+	defer cleanup(t)
+
+	sifSquashImage := filepath.Join(testdir, "data_squash.sif")
+	sifExt3Image := filepath.Join(testdir, "data_ext3.sif")
+	squashfsImage := filepath.Join(testdir, "squashfs.simg")
+	ext3Img := filepath.Join(testdir, "ext3_fs.img")
+
+	// create root directory for squashfs image
+	squashDir, err := os.MkdirTemp(testdir, "root-squash-dir-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Chmod(squashDir, 0o755); err != nil {
+		t.Fatal(err)
+	}
+
+	squashMarkerFile := "squash_marker"
+	if err := fs.Touch(filepath.Join(squashDir, squashMarkerFile)); err != nil {
+		t.Fatal(err)
+	}
+
+	// create the squashfs overlay image
+	cmd := testExec.Command("mksquashfs", squashDir, squashfsImage, "-noappend", "-all-root")
+	if res := cmd.Run(t); res.Error != nil {
+		t.Fatalf("Unexpected error while running command.\n%s", res)
+	}
+
+	// create the overlay ext3 image
+	cmd = testExec.Command("dd", "if=/dev/zero", "of="+ext3Img, "bs=1M", "count=64", "status=none")
+	if res := cmd.Run(t); res.Error != nil {
+		t.Fatalf("Unexpected error while running command.\n%s", res)
+	}
+
+	cmd = testExec.Command("mkfs.ext3", "-q", "-F", ext3Img)
+	if res := cmd.Run(t); res.Error != nil {
+		t.Fatalf("Unexpected error while running command.\n%s", res)
+	}
+
+	// create new SIF images
```

Comment:
Can leave this code commented until SIF images are needed, but we should slightly adapt the test, and avoid running it for now, as it just wastes CI time.